### ISSUE TITLE
useInsertMutation with select returns 400

### ISSUE THREAD

**Issue by AJStorteeboom** [psteinroe/supabase-cache-helpers#527]
useInsertMutation sends a requests with invalid select parameters which causes an error.

I have 2 project tables which is needed for my use case.
```
CREATE TABLE projects (
    id INTEGER NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
    company_id INTEGER NOT NULL REFERENCES public.companies(id) ON DELETE CASCADE,
    project_type TEXT NOT NULL,
    created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
    updated_at TIMESTAMPTZ NOT NULL DEFAULT now()
);

CREATE TABLE nen_projects (
  id INTEGER NOT NULL PRIMARY KEY REFERENCES projects(id) ON DELETE CASCADE,
  company_id INTEGER NOT NULL REFERENCES companies(id) ON DELETE CASCADE,
  year INTEGER,
  number TEXT,
  client TEXT,
  status TEXT DEFAULT 'active',
  created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
  created_by INTEGER NOT NULL REFERENCES profiles(id) ON DELETE SET NULL,
  updated_at TIMESTAMPTZ NOT NULL DEFAULT now(),
  updated_by INTEGER REFERENCES profiles(id) ON DELETE SET NULL,
  soft_deleted        BOOLEAN NOT NULL DEFAULT FALSE,
  soft_deleted_by INTEGER REFERENCES profiles(id) ON DELETE SET NULL,
  soft_deleted_at     TIMESTAMPTZ
);
```

Now I have a form and when I press submit, this is my handler:
```
  const { mutateAsync: createProject, isPending: isCreating } =
    useInsertMutation(supabase.from("projects"), ["id"], "*");
  const { mutateAsync: createProjectNen, isPending: isCreatingNen } =
    useInsertMutation(supabase.from("nen_projects"), ["id"], null);

  const handleCreate = async (values: ProjectFormValues) => {
    try {
      const result = await createProject([{
          project_type: "nen",
          company_id: companyID!,
        }]);
      await createProjectNen([{
          ...values,
          id: result?.[0].id,
          company_id: companyID!,
          created_by: profileID!,
        }]);
      router.push(`/dashboard/projects/${result?.[0].id}`);
    } catch (error) {
      //handle error
    }
  };
```

Now some of the times, the request fails, and I get a 400 error. I am not sure when this happens, but I do know why:

This is the url of the requests that succeeds:
```
http://127.0.0.1:54321/rest/v1/nen_projects?columns="year","number","client"&select=*
```

This is the url of the requests that fails:
```
http://127.0.0.1:54321/rest/v1/nen_projects?columns="year","number","client"&select=*,id,company_id,count,created_at,status
```

which returns body:
```
{
    "code": "42803",
    "details": null,
    "hint": null,
    "message": "aggregate functions are not allowed in RETURNING"
}
```

The select parameter contains 'count' which is an aggregate function. 

Now I have the following questions:

1. Why does useInsertMutation send the count parameter?
2. Am I doing something wrong?
3. How can I prevent this from happening?

My package versions:
```
@supabase-cache-helpers/postgrest-react-query@1.11.1
@supabase-cache-helpers/storage-react-query@1.2.2
@supabase/postgrest-js@1.16.1
@supabase/ssr@0.5.1
@supabase/supabase-js@2.45.4
```
My supabase version: 1.219.2

An important note: This issue is new to me. In my master branch, which has versions, it does NOT occur:
```
@supabase-cache-helpers/postgrest-react-query@1.10.1
@supabase-cache-helpers/storage-react-query@1.2.2
@supabase/postgrest-js@1.17.0
@supabase/ssr@0.5.1
@supabase/supabase-js@2.46.1
```

**Comment by psteinroe** [comment]
Hey, thanks for the issue! So you have any queries in your app on nen_projects? Can you share them?

**Comment by AJStorteeboom** [comment]
Not sure what you mean but is it this:

```
  const {
    data: project = initialData,
    isLoading,
    refetch,
  } = useQuery(
    supabase
      .from("nen_projects")
      .select("*")
      .eq("id", projectId)
      .single(),
    { enabled: false }
  );
```

Or what type of query/code are you looking for?

**Comment by psteinroe** [comment]
I am wondering where `count` is coming from. do you query that anywhere?

**Comment by AJStorteeboom** [comment]
This code executes the requests mentioned in the post.

```
      await createProjectNen([{
          ...values,
          id: result?.[0].id,
          company_id: companyID!,
          created_by: profileID!,
        }]);
```

My bad if it wasn't clear, but it were POST calls to the urls:

```
http://127.0.0.1:54321/rest/v1/nen_projects?columns="year","number","client"&select=*,id,company_id,count,created_at,status
```

![Scherm­afbeelding 2024-11-14 om 18 58 02](https://github.com/user-attachments/assets/28ac38a6-7706-47e0-bd43-bee00bc506d8)

**Comment by AJStorteeboom** [comment]
I looked into the project and I never explicitly execute a count query. The package add's the count.

**Comment by AJStorteeboom** [comment]
I found the issue. After debugging for quite a while I found that I did execute a count query somewhere else but I did it like this:

```
  const { data: totalProjectsCount, isLoading: isLoadingTotalProjectsCount } = useQuery(
    supabase.from("nen_projects").select("count").eq("company_id", companyID!),
    { enabled: true }
  );
```

Which also return's the count. But, the library thinks it is just a regular column which leads to a select of column 'count', which is an aggregate method. So maybe you still want to add a check on the 'count' or any other aggregate function keywords.


For anyone else, I've udpated my count to:
const { isLoading: isLoadingTotalProjects, count: totalProjectsCount } = useQuery(
    supabase.from("nen_projects").select("*",  { count: 'exact' }).eq("company_id", companyID!),
    { enabled: true }
  );
